pyopencl/invoker.py

```diff
@@ -313,7 +313,42 @@ def gen_arg_setting(in_enqueue):
 def generate_enqueue_and_set_args(function_name,
         num_passed_args, num_cl_args,
         arg_types,
-        work_around_arg_count_bug, warn_about_arg_count_bug):
+        work_around_arg_count_bug, warn_about_arg_count_bug, devs):
+
+    # {{{ Check whether argument size exceeds the OpenCL device limit
+
+    if arg_types:
+        for dev in devs:
+            arg_size = 0
+            dev_limit = dev.max_parameter_size
```

Comment:
Note that this will currently cause a lot of warnings in mirgecom, at least until https://github.com/pocl/pocl/pull/1046 is done.

Reply:
You can include logic to catch "if pocl and GPU and max_parameter_size == 1024" to mitigate that.

Reply:
Done in [5addf38](https://github.com/inducer/pyopencl/pull/555/commits/5addf3818f2d145a5cace90c6b2fc7e48e754350)